### PR TITLE
Add extra delete database functionality to DatabaseHandler

### PR DESCRIPTION
**Pull request by corivera**
We already have a drop() method in the base object handler class, but delete has two extra options for closing active connections and deleting backup/restore history.

### CODE REVIEW

**Comment by corivera** [comment]
Still need to add integration tests

File: src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Database/DatabaseHandler.cs [hunk 1]
```diff
@@ -164,14 +163,23 @@ public override Task<InitializeViewResult> InitializeObjectView(InitializeViewRe
                                     EncryptionEnabled = smoDatabase.EncryptionEnabled
                                 };
 
+                                if (!isAzureDB) {
```
Comment (Charles-Gagnon) [inline src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Database/DatabaseHandler.cs:166]: This doesn't seem related to the description, is this just an additional fix or something?

**Review by Charles-Gagnon — APPROVED** [review]

File: src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Database/DatabaseHandler.cs (hunk 1 above)
Comment (corivera) [reply src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Database/DatabaseHandler.cs:166]: Oh, forgot to note that. Yes, it's an additional fix. We don't have the properties dialog enabled for Azure yet, so some of these properties haven't been excluded yet. An error gets thrown when trying to retrieve these properties against an Azure SQL DB.

Comment (corivera) [reply src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Database/DatabaseHandler.cs:166]: Delete _is_ enabled for Azure though, which is why I was hitting it.

Comment (corivera) [reply src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Database/DatabaseHandler.cs:166]: Note: We had a version of delete before for on-prem and azure DBs, but now that it uses a dialog it goes through this initialize view step.

Comment (corivera) [reply src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Database/DatabaseHandler.cs:166]: Already checked with @ssreerama and he aggreed.

File: src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/Contracts/DeleteDatabase.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+#nullable disable
+using Microsoft.SqlTools.Hosting.Protocol.Contracts;
+using Microsoft.SqlTools.Utility;
+
+namespace Microsoft.SqlTools.ServiceLayer.ObjectManagement.Contracts
+{
+    public class DeleteDatabaseRequestParams : GeneralRequestDetails
```
Comment (Charles-Gagnon) [inline src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/Contracts/DeleteDatabase.cs:12]: nit: but I still think drop makes sense to use here to keep the same term usage across STS. So it'd just be `DropDatabaseRequestParams`

But not a huge deal if you'd prefer to keep it like this

Comment (corivera) [reply src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/Contracts/DeleteDatabase.cs:12]: We already have a general Drop method for all object handlers, but I can add an overload for it for databases

Comment (Charles-Gagnon) [reply src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/Contracts/DeleteDatabase.cs:12]: That would ensure that the same logic is always used even if drop is called too which isn't a bad thing.
